script.tag-generator/main.py

```diff
@@ -1,120 +1,141 @@
-import re, urllib2, urllib, httplib, sys, os, time
-import xbmc 
+import datetime
+import json
+import os
+import re
+import sys
+import xbmc
 import xbmcgui 
 import xbmcaddon
-import json
 try:
+    import requests
     import simplejson
     import trakt
     from trakt import users
+    from bs4 import BeautifulSoup
 except:
     if os.name == "nt":
         slash = "\\"
     else:
         slash = "/"
     sys.path.append(os.path.abspath(os.path.dirname(__file__)+slash+"resources"+slash+"lib"))
```

Comment:
> 
> 
> Why do you insert modules on the sys.path if you can just import the module?

Because the dependency is inline and not available in the Kodi repo. My alternative is to put __init__.py in resources and lib, then do a relative import. 


> 
> 
> also use `os.path.join` to join paths without depending on the underlying operating system:
> 
> eg:
> 
> ```python
> os.path.join("firstfolder", "secondfolder", ...)
> ```
Good point. Done.

Reply:
> My alternative is to put `__init__.py` in resources and lib, then do a relative import.

That it how most add-ons are doing it, would be great if you can update it. Changing `sys.path` should be avoided.